pkg/timescale/write.go

```diff
@@ -112,39 +147,37 @@ func contains(s []string, e string) bool {
 }
 
 func insertRowsTimescale(rowsArray []TimescaleRows, config *config.OutputTimescale) error {
-	var c, parseErr = pgx.ParseConnectionString(config.Connection)
-	if parseErr != nil {
-		return parseErr
-	}
 
-	conn, connErr := pgx.Connect(c)
+	conn, connErr := timescalePools[config.Connection].Acquire(context.Background())
```

Comment:
Do not call `context.Background()` repeatedly, create one context and use it throughout the function